US3112 Check off item on list

katherine-boost opened the pull request:
Now when you click on an item on the list, it will toggle the `checked` state between true and false in the database record.
If the list item has `checked: true`, then a CSS class is added which adds a strikethrough.   
(NOTE: it would be ideal if the strikethrough went right along the line, but this isn't straightforward. TODO, pending Emma's design)

Previously, the checked field in the items table was nullable. I've created two migrations:
1. Changing checked from `nil` to `false` in all existing item records
2. Making the `name`, `quantity`, and `checked` fields all not nullable, and defaulting `checked` to false.

<img width="361" alt="shopping-list_strikethrough" src="https://user-images.githubusercontent.com/43123718/45396352-388ad000-b68f-11e8-939b-295611151eac.png">

### Questions

1. Is this an idiomatic way to implement this?
2. So far `ItemsController.update` simply toggles the state. However, in future, there will be other `update` actions, for example incrementing/decrementing the quantity of an item, so I might need some advice on how to structure that.

katherine-boost committed:
US3112 Check off item on list

Now when you click on an item on the list, it will toggle the checked state between true and false in the database record.
If the list item has checked: true, then a CSS class is added which adds a strikethrough. (NOTE: it would be ideal if the strikethrough went right along the line, but this isn't straightforward. TODO!)

Previously, the checked field in the items table was nullable. I've created two migrations:
1. Changing checked from nil to false in all existing item records
2. Making the name, quantity, and checked fields all not nullable, and defaulting checked to false.

katherine-boost commented on app/controllers/items_controller.rb:
```diff
@@ -8,6 +8,13 @@ def create
+    @item = Item.find(params[:id])
+    toggle_state(@item)
+
+    redirect_to list_path(@item.list_id) # TODO: is there a shorthand?
```
Is there an all-in-one path that I should use here?

tim-boost replied:
Not that I can think of, I may have seen the whole model passed to the path generator before but using the id is the most common usage and all good!

tim-boost commented on app/helpers/items_helper.rb:
```diff
@@ -1,2 +1,9 @@
 module ItemsHelper
+  def item_class(item)
+    if item.checked
```
Could use the ternary operator here but completely optional

katherine-boost committed:
US3112 Replace update method for checking item with more specific method

This is because I'll also have another update method for the Items Controller, for incrementing/decrementing the quantity of items.
